src/main/java/seedu/ddd/logic/parser/AddressBookParser.java

```diff
@@ -51,10 +56,32 @@ public Command parseCommand(String userInput) throws ParseException {
         // Lower level log messages are used sparingly to minimize noise in the code.
         logger.fine("Command word: " + commandWord + "; Arguments: " + arguments);
 
+        ArgumentMultimap flagMultimap = ArgumentTokenizer.tokenize(arguments, FLAG_CLIENT, FLAG_VENDOR, FLAG_EVENT);
+        CommandFlag commandFlag = parseFlags(flagMultimap);
+
         switch (commandWord) {
 
-        case AddCommand.COMMAND_WORD:
-            return new AddCommandParser().parse(arguments);
+        case AddContactCommand.COMMAND_WORD:
+
+            if (commandFlag == null) {
+                throw new ParseException(String.format(MESSAGE_INVALID_FLAGS, FLAG_PARSE_ERROR));
+            }
+
+            switch(commandFlag) {
+
+            case CLIENT:
+                return new AddClientCommandParser().parse(arguments);
+
+            case VENDOR:
+                return new AddVendorCommandParser().parse(arguments);
+
+            case EVENT:
```

Comment:
`Event` shouldnt be part of this AddContactCommand case logically even though they use the same command word. Based on the implementation of `AddContactCommand` there is no `Event`. If we are going with the implementation of the interface, perhaps we could declare the keyword inside `AddCommand`?

Reply:
Done.